airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/source.py

```diff
@@ -0,0 +1,407 @@
+            latest_record_date = isoparse(latest_record_date_str)
+            current_parsed_date = isoparse(current_stream_state["date"])
+
+            return {"date": max(current_parsed_date, latest_record_date).isoformat()}
+        else:
+            return {"date": self.start_date.isoformat()}
+
+
+class Transactions(PaypalTransactionStream):
+    """
+    Stream for Transactions /v1/reporting/transactions
+    """
+
+    data_field = "transaction_details"
+    primary_key = ["transaction_info", "transaction_id"]
+    cursor_field = ["transaction_info", "transaction_initiation_date"]
+
+    start_date_max = {"hours": 36}  # this limit is found experimentally
+    records_per_request = 10000  # API limit
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+
+        return "transactions"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+
+        decoded_response = response.json()
+        total_pages = decoded_response.get("total_pages")
+        page_number = decoded_response.get("page")
+        if page_number < total_pages:
+            return {"page": page_number + 1}
+        else:
+            return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        page_number = 1
+        if next_page_token:
+            page_number = next_page_token.get("page")
+
+        return {
+            "start_date": stream_slice["start_date"],
+            "end_date": stream_slice["end_date"],
+            "fields": "all",
+            "page_size": self.page_size,
+            "page": page_number,
+        }
+
+    def stream_slices(
+        self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, any]]]:
+        """
+        Returns a list of slices for each day (by default) between the start date and end date.
+        The return value is a list of dicts {'start_date': date_string, 'end_date': date_string}.
+        """
+        if stream_state and stream_state.get("date"):
+            start_date = isoparse(stream_state.get("date"))
+        else:
+            start_date = self.start_date
+
+        # choose first slice period (for the most recent time)
+        first_slice_period = max(timedelta(**self.start_date_max), timedelta(**self.stream_slice_period))
+        end_date = self.end_date
+
+        start_date_slice = end_date - first_slice_period
+        end_date_slice = end_date
+
+        # Do not run any requests if start_date is less than 36 hrs before current time, otherwise API throws an error:
+        #   'message': 'Data for the given start date is not available.'
+        if start_date > start_date_slice:
```

Comment:
I need help with the decision for this case. The current condition means that API potentially could throw an error mentioned in the comment above. My plan is to return an empty slice which means there will be no queries and no state massages.  Is it a correct way to handle such situations?